pkg/externalresource/broker/file_manager.go

```diff
@@ -0,0 +1,225 @@
+	}
+	if err := os.MkdirAll(res.AbsolutePath(), 0o700); err != nil {
+		return nil, derrors.ErrCreateLocalFileDirectoryFailed.Wrap(err)
+	}
+	// TODO check for quota when we implement quota.
+	return res, nil
+}
+
+// GetPersistedResource checks the given resource exists in the local
+// file system and returns a LocalFileResourceDescriptor.
+func (m *LocalFileManager) GetPersistedResource(
+	creator libModel.WorkerID,
+	resName resModel.ResourceName,
+) (*resModel.LocalFileResourceDescriptor, error) {
+	res := &resModel.LocalFileResourceDescriptor{
+		BasePath:     m.config.BaseDir,
+		Creator:      creator,
+		ResourceName: resName,
+	}
+	if _, err := os.Stat(res.AbsolutePath()); err != nil {
+		if os.IsNotExist(err) {
+			return nil, derrors.ErrResourceDoesNotExist.GenWithStackByArgs(resName)
+		}
+		return nil, derrors.ErrReadLocalFileDirectoryFailed.Wrap(err)
+	}
+
+	m.mu.Lock()
+	defer m.mu.Unlock()
+
+	resources, exists := m.persistedResourcesByCreator[creator]
+	if !exists {
+		return nil, derrors.ErrResourceDoesNotExist.GenWithStackByArgs(resName)
+	}
+	if _, ok := resources[resName]; !ok {
+		return nil, derrors.ErrResourceDoesNotExist.GenWithStackByArgs(resName)
+	}
+
+	return res, nil
+}
+
+// RemoveTemporaryFiles cleans up all temporary files (i.e., unpersisted file resources),
+// created by `creator`.
+func (m *LocalFileManager) RemoveTemporaryFiles(creator libModel.WorkerID) error {
+	log.L().Info("Start cleaning temporary files",
+		zap.String("worker-id", creator))
+
+	creatorResourcePath := filepath.Join(m.config.BaseDir, creator)
+
+	if _, err := os.Stat(creatorResourcePath); err != nil {
+		// The directory not existing is expected if the worker
+		// has never created any local file resource.
+		if os.IsNotExist(err) {
+			log.L().Info("RemoveTemporaryFiles: no local files found for worker",
+				zap.String("worker-id", creator))
+			return nil
+		}
+
+		// Other errors need to be thrown to the caller.
+		return derrors.ErrReadLocalFileDirectoryFailed.Wrap(err)
+	}
+
+	// Iterates over all resources created by `creator`.
+	err := IterOverResourceDirectories(creatorResourcePath, func(resourceID string) error {
+		if m.isPersisted(creator, resourceID) {
+			// Persisted resources are skipped, as they are NOT temporary.
+			return nil
+		}
+
+		fullPath := filepath.Join(m.config.BaseDir, creator, resourceID)
+		if err := os.RemoveAll(fullPath); err != nil {
+			return derrors.ErrCleaningLocalTempFiles.Wrap(err)
+		}
+
+		log.L().Info("temporary resource is removed",
+			zap.String("resource-id", resourceID),
+			zap.String("full-path", fullPath))
+		return nil
+	})
+
+	log.L().Info("Finished cleaning temporary files",
+		zap.String("worker-id", creator))
+	return err
+}
+
+// RemoveResource removes a single resource from the local file system.
+// NOTE the caller should handle ErrResourceDoesNotExist appropriately.
+func (m *LocalFileManager) RemoveResource(creator libModel.WorkerID, resourceID resModel.ResourceName) error {
```

Comment:
Could a worker remove a resource that is not created by itself

Reply:
A worker should be able to `Discard` a resource created by another worker.